data_rentgen/server/utils/lineage_response.py

```diff
@@ -232,3 +230,122 @@ def _get_indirect_column_lineage(column_lineage_by_source_target_id: dict[tuple,
             column_lineage_relation.fields = fields
             relations.append(column_lineage_relation)
     return sorted(relations, key=lambda x: (str(x.from_.id), str(x.to.id)))
+
+
+def _get_datasets(
+    raw_datasets: dict[int, Dataset],
+    outputs: dict[Any, OutputRow],
+    inputs: dict[Any, InputRow],
+) -> dict[str, DatasetResponseV1]:
+    datasets: dict[str, DatasetResponseV1] = defaultdict()
+    for dataset in raw_datasets.values():
+        schema: DatasetSchemaV1 | None = None
+        schema = _get_latest_io_schema(dataset, outputs)
+        if not schema:
+            schema = _get_latest_io_schema(dataset, inputs)
+        datasets[str(dataset.id)] = DatasetResponseV1(
+            id=str(dataset.id),
+            location=LocationResponseV1.model_validate(dataset.location),
+            name=dataset.name,
+            format=dataset.format,
+            schema=schema,
+        )
+    return datasets
+
+
+def _get_latest_io_schema(dataset, relations: Mapping[Any, OutputRow | InputRow]) -> DatasetSchemaV1 | None:
```

Comment:
This could be simplified to:
```python
relations = [relation for relation in relations if relation.dataset_id = dataset_id and relation.schema is not None]
if not relations:
  return None

relations = sorted(relations, key=lambda relation: relation.created_at)
oldest, newest = relations[0], relations[-1]

dataset_schema = DatasetSchemaV1.model_validate(newest_relation.schema)
if oldest.schema_id == newest.schema_id:
  dataset_schema.relevance_type = "EXACT"
else:
  dataset_schema.relevance_type = "LATEST_KNOWN"
return dataset_schema
```